### PR TITLE
[virt_autotest] Add expected_pool_size for virt_autotest

### PR DESCRIPTION
**Pull request by guoxuguang**
- **Description:**
Refer to https://149.44.176.58/tests/5212246, figure out fail at the detached active pool size (57GiB) with the current used pool size 60 GiB on sles15sp3 KVM host.
Should be define a expected pool size both on KVM - 40 GiB and XEN 60 GiB.

- **Related ticket:** 
https://progress.opensuse.org/issues/81679
- **Verification run:** 
[gi-guest_developing-on-host_developing-kvm](http://149.44.176.58/tests/5231528)
[gi-guest_developing-on-host_developing-xen](http://149.44.176.58/tests/5231529)

### CODE REVIEW

**Comment by guoxuguang** [comment]
@alice-suse @Julie-CAO @waynechen55 @nanzhang2020 please help me review this PR.

File: tests/virt_autotest/libvirt_virtual_network_init.pm
Comment (alice-suse) [inline tests/virt_autotest/libvirt_virtual_network_init.pm:54]: What's the actual required disk space before guest installation, and after installation but during virtual_network_init step? They should be different actually, because the installed guests take libvirt/image space a lot, so the value before guest installation should be bigger. In https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11498/files, which checks available disk space before guest installation, the same check value is used as this PR which checks after guest installation. So maybe need another value here.

Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm:54]: Good point ! 
sure, PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11498 had check with disk space for /var/lib/libvirt before guest installation  with `my $expected_libvirt_size = (is_xen_host) ? '60' : '40'; `
Also, we had **defined** (not used ) qcow2 file with 20GiB disk space for each guest installation.
yes, we should follow check with disk space and define qcow2 file to define expected pool size here
for example `my $expected_pool_size = (is_xen_host) ? '40' : '20';` define the enough pool size under /var/lib/libvirt to clone source guest during virtual routed network test.

Comment (alice-suse) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: @guoxuguang I re-checked the code, why the new value is even smaller? The expected_pool_size should be bigger than "the needed disk size for installing guest + additional disk size for virtual network test to clone guest". So it should be BIGGER SIZE before installing guests. And our tests/qam tests uses different guest disk size too, which need different value too.

Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: Great,  imho,  expected_pool_size should be **the deployed guest images(at least KVM-20GiB, XEN-40GiB) + additional disk size(at least 20GiB) for virtual network test to clone guest**, which should be minimum value. if detect current pool value < this expected_pool_size (minimum value), will call` fail_message => "The SUT needs at least " . $expected_pool_size . "GiB available space of active pool for virtual network test"`
So, there was not used with the BIGGER SIZE before our virt_auto test.

Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: Also, regarding qam test, refer to https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/virtualization/universal/prepare_guests.pm, they were used with /var/lib/libvirt/images/xen/ to restore guest images different with us.
```
    assert_script_run "mkdir -p /var/lib/libvirt/images/xen/";

    if (is_sle('<=12-SP1')) {
        script_run "umount /home";
        assert_script_run qq(sed -i 's/\\/home/\\/var\\/lib\\/libvirt\\/images\\/xen/g' /etc/fstab);
        script_run "mount /var/lib/libvirt/images/xen/";
    }
```

Comment (alice-suse) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: > @guoxuguang I re-checked the code, why the new value is even smaller? The expected_pool_size should be bigger than "the needed disk size for installing guest + additional disk size for virtual network test to clone guest". So it should be BIGGER SIZE before installing guests. And our tests/qam tests uses different guest disk size too, which need different value too.

@guoxuguang Sorry, I was wrong on some part. This PR should use SMALLER SIZE indeed because it is a step executed after guest installation , but before virtual nework test. Please set proper value based on ours and qam's different guest size.

**Review by pdostal — CHANGES_REQUESTED** [review]
I'm OK with 20G and also with 40G. We need way more. But can you please move `get_active_pool_and_available_space` and `get_free_mem` to `lib/virt_autotest/virtual_network_utils.pm`?

**Comment by guoxuguang** [comment]
> I'm OK with 20G and also with 40G. We need way more. But can you please move `get_active_pool_and_available_space` and `get_free_mem` to `lib/virt_autotest/virtual_network_utils.pm`?

Great, let me try to move `get_active_pool_and_available_space` and `get_free_mem` to `lib/virt_autotest/virtual_network_utils.pm`.

File: tests/virt_autotest/libvirt_virtual_network_init.pm
Comment (grisu48) [inline tests/virt_autotest/libvirt_virtual_network_init.pm:54]: Perhaps it would be helpful to be able to set the expected pool size via a test SETTING, e.g.

    my $expected_pool_size = get_var('VIRT_EXPECTED_POOLSIZE', (is_xen_host) ? '40' : '20');

Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm:54]: Cool! I will updated this great idea.

File: tests/virt_autotest/libvirt_virtual_network_init.pm
Comment (pdostal) [inline tests/virt_autotest/libvirt_virtual_network_init.pm:54]:
```suggestion
    assert_script_run("test $AVAILABLE_POOL_SIZE -ge $expected_pool_size",
      fail_message => "The SUT needs at least " . $expected_pool_size . "GiB available space of active pool for virtual network test");
```

File: lib/virt_autotest/virtual_network_utils.pm
Comment (pdostal) [inline lib/virt_autotest/virtual_network_utils.pm:43]:
```suggestion
  = qw(download_network_cfg prepare_network restore_standalone destroy_standalone restart_network
     restore_guests restore_network destroy_vir_network restore_libvirt_default enable_libvirt_log pload_debug_log
     check_guest_status check_guest_module check_guest_ip save_guest_ip test_network_interface hosts_backup
     hosts_restore get_free_mem get_active_pool_and_available_space);
```

**Review by pdostal — APPROVED** [review]
Thank you, I think it's good now! Just maybe make the very link lines shorter ;)
Let's also wait for the approval from @grisu48 and @alice-suse and we can merge it.

File: tests/virt_autotest/libvirt_virtual_network_init.pm
Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: Had updated already.

File: tests/virt_autotest/libvirt_virtual_network_init.pm
Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: Had updated already.

File: lib/virt_autotest/virtual_network_utils.pm
Comment (guoxuguang) [reply lib/virt_autotest/virtual_network_utils.pm]: Had updated already.

**Comment by guoxuguang** [comment]
Add the final validation run:
[gi-guest_developing-on-host_developing-kvm](http://149.44.176.58/tests/5308090)

if there were not any new comments and all checks pass, please help us merge this PR.

**Comment by pdostal** [comment]
There is conflicting file @guoxuguang

**Review by grisu48 — APPROVED** [review]

**Comment by guoxuguang** [comment]
> There is conflicting file @guoxuguang

Fix up this conflicting file now.

**Comment by guoxuguang** [comment]
@alice-suse if there was no any comment here, please help me merge this PR. Thanks.

**Pull request merged**
